e2e/template.go

```diff
@@ -576,3 +576,14 @@ type listVMSSVMNetworkInterfaceResult struct {
 		} `json:"properties,omitempty"`
 	} `json:"value,omitempty"`
 }
+
+func extractPrivateIP(res listVMSSVMNetworkInterfaceResult) (string, error) {
+	if res.Value != nil && len(res.Value) > 0 {
```

Comment:
I believe a nil slice will return 0, so the nil check is redundant.

Reply:
hah good to know